server/routes/post-hire-status.ts

```diff
@@ -26,23 +26,38 @@ router.post(
     const user = userId || localUserId;
     // Validate the request body
     await validate(ParticipantPostHireStatusSchema, body);
-    const { participantId } = body;
+    const { participantIds } = body;
     // Check participant exists
-    const [participant] = await getParticipantByID(participantId);
-    if (!participant) {
+    const notParticipants = [];
+    await Promise.all(
+      participantIds.map(async (id) => {
+        const [participant] = await getParticipantByID(id);
+        if (!participant) notParticipants.push(id);
+      })
+    );
```

Comment:
This can be made cleaner and more purely functional with a filter instead of `push`.

```suggestion
    const notParticipants = (
      await Promise.all(participantIds.map(async (id) => ({id, valid: await getParticipantByID(id)).length > 0)})
    ).filter((participant) => participant.valid);
```

Reply:
The push was partly due to the error message getting logged, I wanted to inform which id's didn't have participants
`Participant(s) do not exist with id ${JSON.stringify(notParticipants)}`

Reply:
This example doesn't work as anticipated - We cannot chain a filter onto a Promise.all(), we would require a second variable to filter/map to get the correct format for the alert. 

```
const notParticipants: {
      id: string;
      valid: boolean;
    }[] = await Promise.all(
      participantIds.map(async (id) => ({ id, valid: (await getParticipantByID(id)).length > 0 }))
    );

    const filteredParticipants: {
      id: string;
      valid: boolean;
    }[] = notParticipants.filter((obj) => !obj.valid);

    if (filteredParticipants.length) {
      logger.error({
        action: 'post-hire-status_post',
        message: `Participant(s) do not exist with id **${JSON.stringify(
          filteredParticipants.map(({ id }) => id)
        )}**`,
      });

      return res
        .status(422)
        .send(
          `Participant(s) do not exist with id **${JSON.stringify(
            filteredParticipants.map(({ id }) => id)
          )}.** Please check participant ID`
        );
    }
    ```
    
 Additionally, I can handle the .map in the `filteredParticipants` const vs in JSON.stringify - Although I think this refactor as a whole does not simplify the previous implementation.

Reply:
Discussed Via teams chat. Updated.